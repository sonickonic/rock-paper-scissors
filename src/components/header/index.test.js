import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Header from "./index";
import gameType from "../../gameTypes";

describe("Header", () => {
  const classic = gameType[0];
  it("renders correctly", () => {
    const { asFragment } = render(<Header currentGame={classic} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("calls handleClick prop on logo click", () => {
    const handleClick = jest.fn();
    const { queryByAltText } = render(
      <Header currentGame={classic} handleClick={handleClick} />
    );

    fireEvent.click(queryByAltText("Logo"));

    expect(handleClick).toBeCalledTimes(1);
  });
});
