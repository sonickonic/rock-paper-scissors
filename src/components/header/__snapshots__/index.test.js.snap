// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhCb giPoSW"
  >
    <img
      alt="Logo"
      class="sc-AxhUy fcbIvt"
      src="logo.svg"
    />
    <div
      class="sc-AxjAm cUUMCg"
    >
      <h3
        class="sc-AxirZ fCOrbU"
      >
        score
      </h3>
      <div
        class="sc-AxiKw jcaSpX"
      />
    </div>
  </div>
</DocumentFragment>
`;
